tests/mlmodel_langchain/test_chain.py

```diff
@@ -1841,6 +1845,94 @@ def test_async_langchain_chain_outside_transaction(
     loop.run_until_complete(getattr(runnable, call_function)(input_))
 
 
+@pytest.mark.parametrize(
+    "create_function,call_function,call_function_args,call_function_kwargs,expected_events",
+    (
+        pytest.param(
+            create_structured_output_runnable,
+            "ainvoke",
+            ({"input": "Sally is 13"},),
+            {
+                "config": {
+                    "tags": ["bar"],
+                    "metadata": {"id": "123", "message_ids": ["message-id-0", "message-id-1"]},
+                }
+            },
+            chat_completion_recorded_events_runnable_invoke,
+            id="runnable_chain.ainvoke-with-args-and-kwargs",
+        ),
+        pytest.param(
+            create_structured_output_chain,
+            "ainvoke",
+            ({"input": "Sally is 13"},),
+            {
+                "config": {
+                    "tags": ["bar"],
+                    "metadata": {"id": "123", "message_ids": ["message-id-0", "message-id-1"]},
+                },
+                "return_only_outputs": True,
+            },
+            chat_completion_recorded_events_invoke,
+            id="chain.ainvoke-with-args-and-kwargs",
+        ),
+    ),
+)
+def test_multiple_async_langchain_chain(
+    set_trace_info,
+    json_schema,
+    prompt,
+    chat_openai_client,
+    create_function,
+    call_function,
+    call_function_args,
+    call_function_kwargs,
+    expected_events,
+    loop,
+):
+    call1 = copy.deepcopy(expected_events)
+    call1[0][1]["request_id"] = "b1883d9d-10d6-4b67-a911-f72849704e92"
+    call2 = copy.deepcopy(expected_events)
+    call2[0][1]["request_id"] = "a58aa0c0-c854-4657-9e7b-4cce442f3b61"
+
+    @reset_core_stats_engine()
+    @validate_custom_events(call1 + call2)
+    # 3 langchain events and 5 openai events.
+    @validate_custom_event_count(count=16)
+    @validate_transaction_metrics(
+        name="test_chain:test_multiple_async_langchain_chain.<locals>._test",
+        custom_metrics=[
+            ("Python/ML/Langchain/%s" % langchain.__version__, 1),
+        ],
+        background_task=True,
+    )
+    @background_task()
+    def _test():
+        with patch("langchain_core.callbacks.manager.uuid", autospec=True) as mock_uuid:
```

Comment:
I was trying to see if there was a way to do this without needing to introduce another test dependency through `mock` but this may be fine as is.

Reply:
Yeah as far as I know monkeypatch could be used to override the function but I wouldn't be able to get the side_effect behavior that I have here-it would be a lot more awkward to implement.